File: python/algo/202605/algo3804.py
from typing import List


class Solution:
    def centeredSubarrays(self, nums: List[int]) -> int:
        ans = 0
        n = len(nums)
        
        for i in range(n):
            num_set = set()
            pre_sum = 0
            
            for j in range(i, n):
                pre_sum += nums[j]
                num_set.add(nums[j])
                
                # 如果集合里包含当前前缀和，答案 +1
                if pre_sum in num_set:
                    ans += 1
                    
        return ans        

File: cplus/leetcode/202605/algo3804.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int centeredSubarrays(vector<int>& nums) {
        // 自行解答 暴搜
        int ans = 0, n = nums.size();
        for (int i=0; i<n; i++) {
            unordered_set<int> n_hash; int pres = 0;
            for (int j=i; j<n; j++) {
                pres += nums[j]; n_hash.insert(nums[j]);
                if (n_hash.count(pres)) ans ++;
            }
        }

        return ans;
    }
};